source/AI.cpp

```diff
@@ -2856,6 +2856,25 @@ Point AI::TargetAim(const Ship &ship, const Body &target)
 
 
 
+bool AI::TargetInRange(const Ship &ship, const Hardpoint &hardpoint)
+{
+	shared_ptr<const Body> shipTarget = ship.GetTargetShip();
+	shared_ptr<const Body> targetAsteroid = ship.GetTargetAsteroid();
+	shared_ptr<const Body> target = shipTarget ?: targetAsteroid;
+
+	if(!target)
+		return false;
+
+	Point start = ship.Position() + ship.Facing().Rotate(hardpoint.GetPoint());
+	double distance = start.Distance(target->Position());
+	// Extend the weapon range slightly to account for velocity differences.
```

Comment:
Seems hacky but I guess thats how we usually do it?

Reply:
I think this is done this way else where because you don’t know the future kinetic state of both ships. So any additional calculations would be just as hand wavy.

We could do additional calculations based on current protection positions of the ship and target if their kinetic state stayed the same, but we could still need to buffer that.

Reply:
I mean if its done that way usually we dont have to overhaul it, its fine this way I guess